pkg/service/convert.go

```diff
@@ -141,14 +145,43 @@ func (c *converter) convertResourcePermalinkToName(ctx context.Context, rsc any)
 	return nil
 }
 
-func (c *converter) includeComponentDetail(ctx context.Context, compConfig *componentbase.ComponentConfig, useDynamicDef bool) error {
+func (c *converter) processSetup(ctx context.Context, ownerPermalink string, setup map[string]any) map[string]any {
+	rendered := map[string]any{}
+	for k, v := range setup {
+		switch v := v.(type) {
+		case map[string]any:
+			rendered[k] = c.processSetup(ctx, ownerPermalink, v)
```

Comment:
Question:
I checked archetypeap, huggingface, openai.
They seem to be only 1 layer when we setup those components.
I am not sure if it is common thing to have several map in a setup.
If not common, maybe it is too overkilling?

Reply:
Yeah, we don't have this case for now, but adding it doesn't hurt.